TROUBLESHOOT.md

```diff
@@ -0,0 +1,230 @@
+# Debug Manual Flink - BigQuery Connector
+
+## Overview
+When problems occur, debug tools enable users to quickly pinpoint the root cause, whether it's 
+within the connector itself, the BigQuery API, or the network infrastructure.
+Detailed logs, traces, and debugging capabilities help you effectively troubleshoot issues, 
+reducing the time and effort required to resolve them.
+This document is a debugging manual highlighting possible issues that could be faced by the users 
+and how to troubleshoot them. It also aids in identifying the error causes and proposed steps for 
+mitigating them.
+
+<i>Note: This is a running document containing the issues reported to the Developers of the 
+[Flink - BigQuery Connector](https://github.com/GoogleCloudDataproc/flink-bigquery-connector). 
+Please feel free to contribute as per any new issues that may arise.</i>
+
+## Flink Metrics
+### Writing to BigQuery
+As an effort to increase observability, the Flink-BigQuery Connector Team provides support to 
+collect and report Flink Metrics for a Flink Application.
+The details of the metrics supported so far are available in the 
+[README](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/main/README.md#flink-metrics).
+
+An overview of records seen by the writer and number of records successfully written to 
+BigQuery would enable users to track the flow of records through their application. 
+Comparison of the counts would help troubleshoot if records are being seen by the Sink (Writer) 
+at all, are being serialized and send to the Write API and if BigQuery is able to write 
+these records.
+
+See Flink’s documentation on 
+[Metric Reporters](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/metric_reporters/) 
+to deploy the reporter most conveniently as per user’s need.
+
+## General Debugging
+### Records are not being written to BigQuery
+With the help of metrics available as a part of 0.4.0 release of the connector, 
+users should be able to track the number of records that enter the sink(writer) and the 
+number of records successfully written to BigQuery. 
```

Comment:
lets reference the two metric names here

Reply:
Done.